promql/parser/ast.go

```diff
@@ -133,9 +133,10 @@ type SubqueryExpr struct {
 	// Offset is the offset used during the query execution
 	// which is calculated using the original offset, at modifier time,
 	// eval time, and subquery offsets in the AST tree.
-	Offset    time.Duration
-	Timestamp *int64
-	Step      time.Duration
+	Offset       time.Duration
+	Timestamp    *int64
+	Preprocessor ItemType // START or END.
```

Comment:
While internally, it is helpful to describe this as preprocessed, I'm still confused by calling it `Preprocessor`. It sounds like these items preprocess something. But it's the other way round, those items _get_ preprocessed.

How about `Preprocessed`?

Reply:
For perspective: I see the naming in the code not as critical as how we communicate this to the user. So if you like `Preprocessor` and don't share my concerns about it, never mind. (As said, I assume we'll later converge to a more generic treatment anyway, which will let us get rid of the notion of a preprocessor.)

Reply:
I will name it as `StartOrEnd` to be explicit.